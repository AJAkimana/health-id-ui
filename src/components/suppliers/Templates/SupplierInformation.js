import React, { Fragment } from 'react';
import { Typography, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';

const SupplierInformation = (props) => {
  const {
    classes, renderTextField, supplier
  } = props;
  return (
    <Fragment>
      <div className={classes.dividerDiv}>
        <Typography
          paragraph
          variant="h6"
          align="left"
          gutterBottom
          className={classes.dividerHeaders}
        >
          Supplier Information
        </Typography>
      </div>

      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid item xs={4}>
          {renderTextField(
            classes.descriptionFields, 'tier', 'Tier', supplier.tier.name
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(
            classes.descriptionFields, 'paymentTerms', 'Payment Terms', supplier.paymentTerms.name
          )}
        </Grid>
        <Grid item xs={4}>
          {renderTextField(classes.newTextFields, 'creditDays', 'Credit Days', supplier.creditDays)}
        </Grid>
      </Grid>
    </Fragment>
  );
};

SupplierInformation.propTypes = {
  classes: PropTypes.shape({
    containerGrid: PropTypes.string,
    dividerHeaders: PropTypes.string,
    newTextFields: PropTypes.string,
    descriptionFields: PropTypes.string,
    dividerDiv: PropTypes.string,
  }).isRequired,
  renderTextField: PropTypes.func.isRequired,
  supplier: PropTypes.instanceOf(Object).isRequired
};

export default SupplierInformation;
