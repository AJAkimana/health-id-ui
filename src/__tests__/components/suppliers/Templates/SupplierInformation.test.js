import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SupplierInformation from '../../../../components/suppliers/Templates/SupplierInformation';

const props = {
  supplier: {
    isApproved: false,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Damian Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    creditDays: '3 days',
    notes: [
      {
        id: 1,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 2,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 3,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 4,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 5,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 6,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 7,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 8,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 9,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 10,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      },
      {
        id: 11,
        createdOn: 'Jan 20 2019',
        message: 'Hey bro, nice work',
        createdBy: 'Danilo Silva'
      }
    ]
  },
  renderTextField: jest.fn(),
  variables: {},
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: "Africa/Nairobi"
          }
        }
      }
    }
  }
};

describe('SupplierInformation', () => {
  it('should render the SupplierInformation component properly', () => {
    const setupSupplierInformation = () => {
      const wrapper = shallow(<SupplierInformation {...props} />)
      return wrapper
    }

    let wrapper;
    wrapper = setupSupplierInformation();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
