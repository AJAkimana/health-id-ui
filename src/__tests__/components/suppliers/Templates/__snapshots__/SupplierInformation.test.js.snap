// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierInformation should render the SupplierInformation component properly 1`] = `
<Fragment>
  <div>
    <WithStyles(Typography)
      align="left"
      gutterBottom={true}
      paragraph={true}
      variant="h6"
    >
      Supplier Information
    </WithStyles(Typography)>
  </div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={4}
    />
    <WithStyles(Grid)
      item={true}
      xs={4}
    />
    <WithStyles(Grid)
      item={true}
      xs={4}
    />
  </WithStyles(Grid)>
</Fragment>
`;
